lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2494,7 +2836,39 @@ class PrimalGenCloner final
     mapValue(ai, originalDirectResult);
 
     // Checkpoint the pullback.
-    getPrimalInfo().addPullbackDecl(ai, pullback->getType());
+    auto *pullbackDecl =
+        getPrimalInfo().addPullbackDecl(ai, getOpType(pullback->getType()));
+
+    // If actual pullback type does not match lowered pullback type, reabstract
+    // the pullback using a thunk.
+    auto actualPullbackType =
+        getOpType(pullback->getType()).getAs<SILFunctionType>();
+    auto canGenSig = SubsMap.getGenericSignature()
+        ? SubsMap.getGenericSignature()->getCanonicalSignature()
+        : nullptr;
+    Lowering::GenericContextScope genericContextScope(
+        context.getTypeConverter(), canGenSig);
+    auto loweredPullbackType =
+        getOpType(context.getTypeConverter().getLoweredType(
+                      pullbackDecl->getInterfaceType()->getCanonicalType()))
+            .castTo<SILFunctionType>();
+    if (!loweredPullbackType->isEqual(actualPullbackType)) {
+      // Set original pullback type before reabstraction.
+      // NOTE: Is there a better `DenseMap` API?
```

Comment:
@rxwei Is there a shorter `DenseMap` API than this super long call here?
I just want to update the `NestedApplyInfo.originalPullbackType` field.

Reply:
Answered in a separate review comment because i didn't see this.